index.js

```diff
@@ -22,5 +27,15 @@ module.exports = function (grunt, opts) {
 		return result.concat(Array.isArray(deps) ? deps : Object.keys(deps));
 	}, []);
 
-	multimatch(names, pattern).forEach(grunt.loadNpmTasks);
+	var resolver = resolveFrom.bind(null, process.cwd());
+
+	multimatch(names, pattern).forEach(function (pkgName) {
+		if (config.nodeModulesResolution) {
+			var resolved = resolver(path.join(pkgName, 'package.json'));
+			var tasksDir = path.join(path.dirname(resolved), 'tasks');
+			grunt.loadTasks(tasksDir);
```

Comment:
Also I prefer to not reassign variables, especially those passed in as arguments. But I'll change the if statement.